.github/workflows/upload-and-deploy.yaml

```diff
@@ -24,45 +24,36 @@ jobs:
     name: Upload files to S3 bucket
     runs-on: ubuntu-latest
     needs: pre-commit
-    if: startsWith(github.ref, 'refs/pull') != true
+    if: github.event_name == 'push' #startsWith(github.ref, 'refs/pull') != true
     environment: develop
     steps:
 
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Set up python 3.8
-        uses: actions/setup-python@v2
+      - name: Setup code, pipenv, aws
+        uses: Sage-Bionetworks/action-pipenv-aws-setup@v1
         with:
-          python-version: 3.8
+          aws_access_key_id: ${{ secrets.CI_USER_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.CI_USER_SECRET_ACCESS_KEY }}
+          role_to_assume: ${{ secrets.CI_ROLE_TO_ASSUME }}
 
       - name: Setup sam
         uses: aws-actions/setup-sam@v1
         with:
           version: 1.37.0
 
-      - name: Assume AWS role
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.CI_USER_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.CI_USER_SECRET_ACCESS_KEY }}
-          aws-region: us-east-1
-          role-to-assume: ${{ secrets.CI_ROLE_TO_ASSUME }}
-          role-duration-seconds: 1200
+      - name: Set namespace for main
+        if: github.ref_name == 'main'
+        run: echo "NAMESPACE=bridge-downstream" >> $GITHUB_ENV
```

Comment:
Why create this exception for the main branch? Seems like it would be more intuitive to simply use "main" as the namespace.

Reply:
This leads me to think about the generalization of this.  If we have NAMESPACES as the github branch, if we had other projects that utilized this paradigm, would we have competing cloudformation stack names?

Reply:
The namespace is used to name the stack. I simply didn't like my stacks named "main-glue-foojob".  "bridge-downstream-glue-foojob" is more meaningful and official sounding to me. "ETL-121-glue-foojob" sounds to me like a stack that is in development. That was my reasoning.

`NAMESPACE` is being deliberately overloaded: it is being used as a stack prefix, and it is being used to construct artifact paths for S3. I deliberately overloaded it so we wouldn't have to pass two different things, both an acceptable`stack-prefix` variable and a `namespace` variable. We might even run the danger that those turn into four different things.

If you are adding a stack and you are merging it to main and it should be a unique stack, then it will get a unique name.